lib/os/shared_multi_heap.c

```diff
@@ -1,118 +0,0 @@
-/*
- * Copyright (c) 2021 Carlo Caione, <ccaione@baylibre.com>
- *
- * SPDX-License-Identifier: Apache-2.0
- */
-
-#include <zephyr/kernel.h>
-#include <zephyr/device.h>
-#include <zephyr/sys/sys_heap.h>
-#include <zephyr/sys/multi_heap.h>
-
-#include <zephyr/multi_heap/shared_multi_heap.h>
-
-static struct sys_multi_heap shared_multi_heap;
```

Comment:
Pedantry: are we absolutely sure this doesn't have any out-of-tree users?  I'll let the experts chime in on whether this needs formal deprecation or not, just calling it out.  I personally don't mind the deletion.

Reply:
Apparently we have at least one user (@talih0) so I'll leave that in place so that we don't need to deprecate anything.